docs/encyclopedia/application-message-passing.mdx

```diff
@@ -220,22 +220,40 @@ Every time the Workflow wakes up--generally, it wakes up when it needs to--it wi
 
 This execution is on a single thread–while this means you don’t have to worry about parallelism, you do need to worry about concurrency if you have written Signal and Update handlers that can block. These can run interleaved with the main Workflow and with one another, resulting in potential race conditions. These methods should be made reentrant.
 
+#### Initializing the Workflow first {#workflow-initializers}
+
+Initialize your Workflow's state before handling messages. This prevents your Workflow reading uninitialized instance variables.
```

Comment:
I think you mean that it prevents your Handlers from reading uninitialized instance variables. 
```suggestion
Initialize your Workflow's state before handling messages.
This prevents your Handlers reading uninitialized instance variables if they arrive before the Workflow state is entire set up.
```

Reply:
Workflow is correct but handler is more precise.